readthedocs/projects/tasks.py

```diff
@@ -289,8 +292,11 @@ def run(self, version_pk):  # pylint: disable=arguments-differ
     ),
 )
 def update_docs_task(self, project_id, *args, **kwargs):
-    step = UpdateDocsTaskStep(task=self)
-    return step.run(project_id, *args, **kwargs)
+    try:
+        step = UpdateDocsTaskStep(task=self)
+        return step.run(project_id, *args, **kwargs)
+    finally:
+        clean_build_task(kwargs.get('version_pk'))
```

Comment:
So, to make sure we always have a `version_pk` we need https://github.com/rtfd/readthedocs.org/pull/5765